source/NFSProStreet.GenericFix/dllmain.cpp

```diff
@@ -620,52 +620,43 @@ void Init()
         }; injector::MakeInline<DeadzoneHookY>(pattern.get_first(18 + 0), pattern.get_first(18 + 6));
     }
 
-    auto GetFolderPathpattern = hook::pattern("50 6A 00 6A 00 68 ? 80 00 00 6A 00");
-    if (bWriteSettingsToFile && injector::GetBranchDestination(GetFolderPathpattern.get(0).get<uintptr_t>(14), true).as_int() == 0)
+    if (bWriteSettingsToFile)
     {
-        auto msgboxID = MessageBox(
-                            NULL,
-                            (LPCWSTR)L"WriteSettingsToFile option will not work with your exe version. Use different exe or disable that option.\nDo you want to disable it now?",
-                            (LPCWSTR)L"NFSProStreet.GenericFix",
-                            MB_ICONWARNING | MB_YESNO | MB_DEFBUTTON1
-                        );
-
-        if (msgboxID == IDYES)
+      struct LazyHook
+      {
+        static void hook()
         {
-            iniReader.WriteInteger("MISC", "WriteSettingsToFile", 0);
-            bWriteSettingsToFile = false;
-        }
-    }
+          auto GetFolderPathpattern = hook::pattern("50 6A 00 6A 00 68 ? 80 00 00 6A 00");
+          while (!injector::GetBranchDestination(GetFolderPathpattern.get(0).get<uintptr_t>(14), true).as_int())
+            std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
Is this ok though? What if the game uses unpatched function during that sleep and then patches it?

Reply:
The wait call is really only there for extreme cases. The instant all scripts are loaded from the ASI Loader, syscalls are defined.

Also, if an absolute patch is necessary, it is possible to detour the call instruction itself.